lib/src/local_working_copy.rs

```diff
@@ -1085,15 +1111,17 @@ impl FileSnapshotter<'_> {
 
                     // If the whole directory is ignored, visit only paths we're already
                     // tracking.
-                    self.visit_tracked_files(file_states)?;
-                } else {
+                    self.spawn_ok(scope, move |_| self.visit_tracked_files(file_states));
```

Comment:
comment (non-blocking): Nice, I completely neglected/didn't consider that we're blocking on the sub-directory job here.